libtsuba/src/RDG.cpp

```diff
@@ -385,14 +385,15 @@ tsuba::RDG::DoMake(const katana::Uri& metadata_dir) {
       return part_result.error();
     }
 
-    if (local_to_global_id_ != nullptr) {
-      if (local_to_user_id_ == nullptr) {
-        // for backward compatibility
-        // NB: this is a zero-copy slice, so the underlying data is shared
-        set_local_to_user_id(local_to_global_id_->Slice(0));
-      } else if (local_to_user_id_->length() != local_to_global_id_->length()) {
+    if (local_to_user_id_->length() == 0) {
+      // for backward compatibility
+      // NB: this is a zero-copy slice, so the underlying data is shared
+      set_local_to_user_id(local_to_global_id_->Slice(0));
+    } else if (local_to_user_id_->length() != local_to_global_id_->length()) {
+      if (local_to_user_id_->length() != local_to_global_id_->length()) {
```

Comment:
This if statement seems redundant.

Reply:
I'm out of my depth here.  I added what Roshan told me to add, and it seems rational to only set local_to_user_id if it wasn't set by the RDG.  But @roshandathathri, can you comment?  

In general, in this file, I see these chunked arrow arrays being tested for nullptr.  The lifecycles of RDGs are hard to understand (IMHO), so maybe these array become null when the RDG is unbound from storage.  Or maybe it was because they were initialized to nullptr.  But because Distribution assumes these arrays are not nullptr, I now initialize them in RDG's constructor.  We could change the code Distribution calls to check for nullptr, but I am always in favor of eliminating nullptr checks if we can.  In this case arrow forces us to use std::shared_ptr, but I think programmers (perhaps unwittingly) assume the member variables are initialized.

Reply:
The same condition is checked twice inadvertently:
```cpp
else if (local_to_user_id_->length() != local_to_global_id_->length()) {
  if (local_to_user_id_->length() != local_to_global_id_->length()) {
```

Reply:
Superceeded by #141. 🤦